std/string.d

```diff
@@ -350,15 +350,22 @@ alias CaseSensitive = Flag!"caseSensitive";
         If the parameters are not valid UTF, the result will still
         be in the range [-1 .. s.length], but will not be reliable otherwise.
   +/
-ptrdiff_t indexOf(Range)(Range s, in dchar c,
+ptrdiff_t indexOf(Range)(Range sIn, in dchar c,
         in CaseSensitive cs = CaseSensitive.yes)
-    if (isInputRange!Range && isSomeChar!(ElementEncodingType!Range))
+    if ( (isInputRange!Range && isSomeChar!(ElementEncodingType!Range)) ||
+        __traits(compiles, StringTypeOf!Range)
+    )
 {
     import std.ascii : toLower, isASCII;
     import std.uni : toLower;
     import std.utf : byDchar, byCodeUnit, UTFException, codeLength;
     alias Char = Unqual!(ElementEncodingType!Range);
 
+    static if (__traits(compiles, StringTypeOf!Range))
+        StringTypeOf!Range s = sIn;
```

Comment:
it is new in std.traits and tells you to what kind of string the struct or class can be assigned to